Ginger/GingerCore/ALM/JIRA/Bll/JiraExportManager.cs

```diff
@@ -47,15 +47,85 @@ public JiraExportManager(JiraRepository.JiraRepository jiraRep)
+                    {
+                        foreach (DefectStatus defectStatus in jiraDefectStatuses)
+                        {
+                            ComparisionOperator comparisionOperator = ComparisionOperator.EqualTo;
+                            if (defectStatus.Operator.ToLower().Contains("not"))
+                            {
+                                comparisionOperator = ComparisionOperator.NotEqual;
+                            }
+                            filterData.Add(new WhereData() { Name = "Status", Values = new List<string>() { defectStatus.Status }, Operator = WhereOperator.And, ComparisionOperator = comparisionOperator });
+                        }
+                    }
+
+
+                    var issues = jiraRepObj.GetJiraIssues(ALMCore.DefaultAlmConfig.ALMUserName, ALMCore.DefaultAlmConfig.ALMPassword, ALMCore.DefaultAlmConfig.ALMServerURL, ALMCore.DefaultAlmConfig.ALMProjectKey, ResourceType.DEFECT, filterData);
+                    if (issues.DataResult.Count > 0)
+                    {
+                        //Add existing defect id
+                        defectsOpeningResults.Add(defectForOpening.Key, issues.DataResult.FirstOrDefault().id.ToString());
+                        continue;
+                    }
+                    else
+                    {
+                        string paths = defectForOpening.Value.ContainsKey("screenshots") ? defectForOpening.Value["screenshots"] : string.Empty;
+                        screenshots.Add(paths);
+                    }
+                }
+
+                //if no then add into list to open new defect
                 defectsToExport.Add(this.CreateDefectData(defectForOpening, defectsFields));
             }
-            var exportedDefects = jiraRepObj.ExportJiraIssues(ALMCore.DefaultAlmConfig.ALMUserName, ALMCore.DefaultAlmConfig.ALMPassword, ALMCore.DefaultAlmConfig.ALMServerURL, defectsToExport);
-            for (var a = 0; a < exportedDefects.Count; a++)
+
+            if(defectsToExport.Count > 0)
             {
-                if (defectsForOpening.Count > a)
-                    defectsOpeningResults.Add(defectsForOpening.ElementAt(a).Key, exportedDefects[a].DataResult.key);
+                //Create new Defect
+                var exportedDefects = jiraRepObj.ExportJiraIssues(ALMCore.DefaultAlmConfig.ALMUserName, ALMCore.DefaultAlmConfig.ALMPassword, ALMCore.DefaultAlmConfig.ALMServerURL, defectsToExport);
+                if (exportedDefects.Count > 0)
+                {
+                    for (var a = 0; a < exportedDefects.Count; a++)
+                    {
+                        if (exportedDefects[a].DataResult != null)
+                        {
+                            //Add attachments to opened defects
+                            string[] screenshotsPaths = screenshots[a].Split(new string[] { "," }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6043077)